x-pack/plugins/enterprise_search/server/utils/create_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { formatMlPipelineBody } from './create_pipeline_definitions';
+
+/**
+ * Indicators for a possibly created pipeline.
+ */
+export interface CreatedPipelineFlags {
+  created?: boolean;
+  exists?: boolean;
+}
+
+/**
+ * Creates a Machine Learning Inference pipeline with the given settings, if it doesn't exist yet.
+ * @param pipelineName pipeline name set by the user.
+ * @param modelId model ID selected by the user.
+ * @param sourceField The document field that model will read.
+ * @param destinationField The document field that the model will write to.
+ * @param esClient the Elasticsearch Client to use when retrieving model details.
+ */
+export const createMlInferencePipeline = async (
+  pipelineName: string,
+  modelId: string,
+  sourceField: string,
+  destinationField: string,
+  esClient: ElasticsearchClient
+): Promise<CreatedPipelineFlags> => {
+  const inferencePipelineGeneratedName = `ml-inference-${pipelineName}`;
+
+  // Check that a pipeline with the same name doesn't already exist
+  try {
+    const pipelineByName = await esClient.ingest.getPipeline({
+      id: inferencePipelineGeneratedName,
+    });
+
+    if (pipelineByName[inferencePipelineGeneratedName]) {
```

Comment:
Suggestion: The pattern we follow here is we [throw an error](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/server/lib/connectors/add_connector.ts#L51) with an error code and provide a [better localised error message as response](https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/server/routes/enterprise_search/connectors.ts#L42).

Reply:
@joemcelroy Refactored to throw error and handle it with `createError()` in the API layer, please review.
I'll add a tech debt ticket for localization in order to keep the scope of this PR small.